### PR TITLE
Service refactoring

### PR DESCRIPTION
**Pull request by atanasdinov**
# Description

## What

Modernizes the existing codebase.

## Why

[JIRA Ticket](https://financialtimes.atlassian.net/browse/UPPSF-3330)

## Anything, in particular, you'd like to highlight to reviewers

TODO

## Scope and particulars of this PR (Please tick all that apply)

- [x] Tech hygiene (dependency updating & other tech debt)
- [x] Bug fix
- [x] Feature
- [x] Documentation
- [ ] Breaking change
- [ ] Minor change (e.g. fixing a typo, adding config)

___
This Pull Request follows the rules described in our [Pull Requests Guide](https://github.com/Financial-Times/upp-docs/tree/master/guides/pr-guide)

### CODE REVIEW

File: mongo/inquirer.go [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+
+type contentFinder interface {
+	findContent(ctx context.Context, candidates []string) (cursor, int, error)
+}
+
+type cursor interface {
+	Next(ctx context.Context) bool
+	Decode(val interface{}) error
+	Err() error
+
```
Comment (dtvalk-ov) [inline mongo/inquirer.go:20]: Is that gap intentional?

File: mongo/inquirer.go [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+func (i *Inquirer) processDocuments(ctx context.Context, cur cursor, docs chan *content.Stub) {
+	defer func() {
+		close(docs)
+		_ = cur.Close(context.Background())
+	}()
+
+	counter := 0
+
+	for cur.Next(ctx) {
```
Comment (dtvalk-ov) [inline mongo/inquirer.go:56]: **Nitpick and suggestion**: maybe rename this to c or cursor? 😅

File: queue/listener.go [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+	messageConsumer *kafka.Consumer
+	locker          *export.Locker
+	sync.RWMutex
+	paused              bool
+	terminator          *export.Terminator
+	received            chan *Notification
+	pending             map[string]*Notification
+	notificationHandler *NotificationHandler
+	messageMapper       *MessageMapper
+	worker              chan struct{}
```
Comment (dtvalk-ov) [inline queue/listener.go:22]: Isn't it more appropriate to name it worker**s**?

File: queue/listener.go [hunk 4]
```diff
@@ -0,0 +1,222 @@
...
+		return "", err
+	}
+	return "Kafka is good to go.", nil
+}
+
+func (l *Listener) MonitorCheck() (string, error) {
+	if err := l.messageConsumer.MonitorCheck(); err != nil {
+		return "", err
+	}
+	return "Kafka is good to go.", nil
```
Comment (dtvalk-ov) [inline queue/listener.go:221]: Shouldn't the 2 separate kafka checks have different healthy status strings? Like _Connectivity to kafka consumer is OK_ for the connectivity check and _Kafka consumer status is OK_ for the monitor checlk

File: mongo/inquirer.go (hunk 1 above)
Comment (atanasdinov) [reply mongo/inquirer.go:20]: Kinda yes, thought it looks better to have the "functionality" split from the lifecycle.

**Review by dtvalk-ov — DISMISSED** [review]

**Review by martin-stanchev — APPROVED** [review]
LGTM

**Review by dtvalk-ov — DISMISSED** [review]

**Review by martin-stanchev — DISMISSED** [review]

**Review by martin-stanchev — APPROVED** [review]

**Review by dtvalk-ov — APPROVED** [review]

**Pull request merged**